lib/ansible/executor/process/worker.py

```diff
@@ -125,7 +132,36 @@ def _hard_exit(self, e):
 
         os._exit(1)
 
-    def run(self):
+    def _detach(self) -> None:
+        """
+        The intent here is to detach the child process from the inherited stdio fds,
+        including /dev/tty. Children should use Display instead of direct interactions
+        with stdio fds.
+        """
+        try:
+            os.setsid()
+            # Create new fds for stdin/stdout/stderr, but also capture python uses of sys.stdout/stderr
+            for fd, mode in (
+                    (STDIN_FILENO, os.O_RDWR | os.O_NONBLOCK),
+                    (STDOUT_FILENO, os.O_WRONLY),
```

Comment:
In testing I ran into an issue with a deadlock on flushing, although that was when not setting `sys.stdout` and `sys.stderr` to `io.StringIO`.  So it is potentially feasible, so I can attempt to validate. It will still require 2 calls to `dup2` for explicitly setting the correct FDs.